File: library/src/main/java/wannabe/Voxel.java
package wannabe;

public class Voxel {
  public final Position position;
  public final int value;

  public Voxel(int x, int y, int z, int color) {
    this(new Position(x, y, z), color);
  }

  public Voxel(Position position, int color) {
    this.position = position;
    this.value = color;
  }
}

File: library/src/main/java/wannabe/Voxel.kt
package wannabe

/** A value with a 3d position.  */
data class Voxel(
  val position: Position,
  /** Value of this voxel, could represent a texture or a color.  */
  val value: Int
) {

  constructor(
    x: Int,
    y: Int,
    z: Int,
    value: Int
  ) : this(Position(x, y, z), value)

}
